frontend/src/screens/internal-apps/ZapPlanner.tsx

```diff
@@ -131,6 +237,34 @@ export function ZapPlanner() {
       };
 
       const createAppResponse = await createApp(createAppRequest);
+      // months → days since months are not recognized
+      const monthsToDays = (m: string) => parseInt(m, 10) * 31;
+
+      const sleepDuration =
+        frequencyUnit === "months"
+          ? `${monthsToDays(frequencyValue)} days`
+          : `${frequencyValue} ${frequencyUnit}`;
+
+      // Build a “stable fiat” payload when needed
+      const subscriptionBody: Record<string, unknown> = {
+        recipientLightningAddress,
+        message: comment || "ZapPlanner payment from Alby Hub",
+        payerData: JSON.stringify({
+          ...(senderName ? { name: senderName } : {}),
+        }),
+        nostrWalletConnectUrl: createAppResponse.pairingUri,
+        sleepDuration,
+      };
+
+      if (amountCurrency === "BTC") {
+        // user entered sats directly
+        subscriptionBody.amount = satsToSend;
+        subscriptionBody.currency = "BTC";
+      } else {
+        // user entered fiat → ZapPlanner will convert to sats later
+        subscriptionBody.amount = parseFloat(amount);
```

Comment:
I think unfortunately only whole numbers are supported right now. I tried 0.5 USD and it returned a 400 error.